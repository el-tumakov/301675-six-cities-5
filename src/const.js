export const FEATURES = [
  `Wi-Fi`,
  `Washing machine`,
  `Towels`,
  `Heating`,
  `Coffee machine`,
  `Baby seat`,
  `Kitchen`,
  `Dishwasher`,
  `Cabel TV`,
  `Fridge`
];

export const OFFERS_TYPE = [
  `apartment`,
  `room`,
  `house`,
  `hotel`
];
