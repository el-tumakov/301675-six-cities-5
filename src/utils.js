export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const adaptOffersToClient = (offers) => {
  const adaptedOffers = offers.map((offer) => {
    const adaptedOffer = Object.assign(
        {},
        offer,
        {
          owner: {
            id: offer.host.id,
            avatar: offer.host.avatar_url,
            name: offer.host.name,
            super: offer.host.is_pro
          },
          photos: offer.images,
          favorite: offer.is_favorite,
          premium: offer.is_premium,
          guests: offer.max_adults,
          preview: offer.preview_image,
          features: offer.goods
        }
    );

    delete adaptedOffer.host;
    delete adaptedOffer.images;
    delete adaptedOffer.is_favorite;
    delete adaptedOffer.is_premium;
    delete adaptedOffer.max_adults;
    delete adaptedOffer.preview_image;
    delete adaptedOffer.goods;


    return adaptedOffer;
  });

  return adaptedOffers;
};

export const adaptReviewsToClient = (reviews) => {
  const adaptedReviews = reviews.map((review) => {
    const adaptedReview = Object.assign(
        {},
        review,
        {
          user: Object.assign({}, review.user, {
            avatar: review.user.avatar_url,
            isPro: review.user.is_pro
          })
        }
    );

    delete adaptedReview.user.avatar_url;
    delete adaptedReview.user.is_pro;

    return adaptedReview;
  });

  return adaptedReviews;
};
