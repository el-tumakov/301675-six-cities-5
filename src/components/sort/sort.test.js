import React from "react";
import renderer from "react-test-renderer";
import Sort from "./sort";
import {noop} from "../../utils";
import {SortTypes} from "../../const";


it(`Should Sort render correctly`, () => {
  const tree = renderer
      .create(
          <Sort
            activeSort={SortTypes.POPULAR}
            onChangeSort={noop}
            isSortOpened={true}
            onSortFocus={noop}
            onSortBlur={noop}
          />
      ).toJSON();

  expect(tree).toMatchSnapshot();
});
