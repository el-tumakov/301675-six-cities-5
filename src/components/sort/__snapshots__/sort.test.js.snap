// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Sort render correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="places__sorting-type"
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--closed"
  >
    <li
      className="places__option places__option--active"
      onMouseDown={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      onMouseDown={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      onMouseDown={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      onMouseDown={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
  <select
    className="places__sorting-type visually-hidden"
    defaultValue="popular"
    id="places-sorting"
  >
    <option
      className="places__option"
      value="popular"
    >
      Popular
    </option>
    <option
      className="places__option"
      value="to-low"
    >
      Price: low to high
    </option>
    <option
      className="places__option"
      value="to-high"
    >
      Price: high to low
    </option>
    <option
      className="places__option"
      value="top-rated"
    >
      Top rated first
    </option>
  </select>
</form>
`;
